x-pack/plugins/security_solution/public/entity_analytics/pages/entity_store_management_page.tsx

```diff
@@ -0,0 +1,456 @@
+    <EuiPanel hasBorder={true} paddingSize="l" grow={false}>
+      <EuiFlexGroup alignItems="center" gutterSize="s">
+        <EuiIcon type="questionInCircle" size="xl" />
+        <EuiTitle size="xxs">
+          <h3>
+            <FormattedMessage
+              id="xpack.securitySolution.entityAnalytics.assetCriticalityUploadPage.information.title"
+              defaultMessage="What is asset criticality?"
+            />
+          </h3>
+        </EuiTitle>
+      </EuiFlexGroup>
+      <EuiSpacer size="s" />
+      <EuiText size="s">
+        <FormattedMessage
+          id="xpack.securitySolution.entityAnalytics.assetCriticalityUploadPage.information.description"
+          defaultMessage="Asset criticality allows you to classify entities based on their importance and impact on business operations. Use asset criticality to guide prioritization for alert triaging, threat-hunting, and investigation activities."
+        />
+      </EuiText>
+      <EuiHorizontalRule />
+      <EuiTitle size="xxs">
+        <h4>
+          <FormattedMessage
+            id="xpack.securitySolution.entityAnalytics.assetCriticalityUploadPage.information.usefulLinks"
+            defaultMessage="Useful links"
+          />
+        </h4>
+      </EuiTitle>
+      <EuiSpacer size="xs" />
+
+      <EuiLink
+        target="_blank"
+        rel="noopener nofollow noreferrer"
+        href={entityAnalyticsLinks.assetCriticality}
+      >
+        <FormattedMessage
+          id="xpack.securitySolution.entityAnalytics.assetCriticalityUploadPage.documentationLink"
+          defaultMessage="Asset criticality documentation"
+        />
+      </EuiLink>
+    </EuiPanel>
+  );
+};
+
+const EntityStoreFeatureFlagNotAvailableCallout: React.FC = () => {
+  return (
+    <>
+      <EuiSpacer size="m" />
+      <EuiCallOut
+        title={
+          <FormattedMessage
+            id="xpack.securitySolution.entityAnalytics.entityStoreManagementPage.featureFlagDisabled"
+            defaultMessage="Entity Store capabilities not available"
+          />
+        }
+        color="primary"
+        iconType="iInCircle"
+      >
+        <EuiText size="s">
+          <FormattedMessage
+            id="xpack.securitySolution.entityAnalytics.assetCriticalityUploadPage.entityStoreManagementPage.featureFlagDisabledDescription"
+            defaultMessage="The full capabilities of the Entity Store have been disabled in this environment. Contact your administrator for further assistance."
+          />
+        </EuiText>
+      </EuiCallOut>
+    </>
+  );
+};
+
+const EntityStoreHealth: React.FC<{ currentEntityStoreStatus: string }> = ({
+  currentEntityStoreStatus,
+}) => {
+  return (
+    <EuiHealth
+      textSize="m"
+      color={entityStoreEnabledStatuses.includes(currentEntityStoreStatus) ? 'success' : 'subdued'}
+    >
+      {entityStoreEnabledStatuses.includes(currentEntityStoreStatus) ? 'On' : 'Off'}
+    </EuiHealth>
+  );
+};
+
+const EnablementButton: React.FC<{
```

Comment:
You could wrap this in a memo to avoid re-rendering unless the props change.

Reply:
As per the react docs: 
> Optimizing with `memo` is only valuable when your component re-renders often with the same exact props, and its re-rendering logic is expensive. If there is no perceptible lag when your component re-renders, memo is unnecessary

That doesn't really apply here, as most of the page's re-renders will be due to some state change that impacts what the button displays.
Plus, I already think we have too many unecessary `memo`s in Kibana as it is :sweat_smile:

Reply:
As per the react docs: 
> Optimizing with `memo` is only valuable when your component re-renders often with the same exact props, and its re-rendering logic is expensive. If there is no perceptible lag when your component re-renders, memo is unnecessary

That doesn't really apply here, as most of the page's re-renders will be due to some state change that impacts what the button displays.
Plus, I already think we have too many unecessary `memo`s in Kibana as it is :sweat_smile: